### ISSUE TITLE
Error message for database access in setup_X is misleading

### ISSUE THREAD

**Issue by pelme** [pytest-dev/pytest-django#297]
The error message when triggering the database from setup_X methods should clearly point out that database access from such methods is not supported, rather than advising to put the django_db marker on the test.

See http://stackoverflow.com/questions/34089425/django-pytest-setup-method-database-issue for original problem.

**Comment by TauPan** [comment]
May I also suggest some documentation, similar to what @pelme has written in his very helpful SO answer. At least I couldn't find that setup-methods unsupported in the pytest-django documentation anywhere.

**Comment by pelme** [comment]
Yes that would be very helpful. If you (or anyone else) are interested in contributing a documentation change based on my SO answer, that would be very helpful!
